connectors/src/connectors/notion/lib/upsert.ts

```diff
@@ -0,0 +1,55 @@
+import axios, { AxiosRequestConfig, AxiosResponse } from "axios";
+
+import { errorFromAny } from "@connectors/lib/error";
+import logger from "@connectors/logger/logger";
+import { DataSourceConfig } from "@connectors/types/data_source_config";
+
+const DUST_API_URL = process.env.DUST_API_URL;
+if (!DUST_API_URL) {
+  throw new Error("DUST_API_URL not set");
+}
+
+export async function upsertToDatasource(
```

Comment:
should this be a shared function across workers?

Reply:
> should this be a shared function across workers? 
I can move it to `@connectors/lib` 👍 not going to update the slack code to avoid rebase problems

Reply:
done